crates/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -94,6 +94,8 @@ pub enum TypeCheckError {
     },
     #[error("Cannot infer type of expression, type annotations needed before this point")]
     TypeAnnotationsNeeded { span: Span },
+    #[error("use of deprecated function {name}")]
+    CallDeprecated { name: String, span: Span },
```

Comment:
We want to be able to provide an additional message like "go use function X instead"